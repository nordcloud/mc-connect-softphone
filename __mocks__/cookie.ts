export default {
  serialize: jest.fn().mockReturnValue('dummyCookieSerializeValue'),
};
