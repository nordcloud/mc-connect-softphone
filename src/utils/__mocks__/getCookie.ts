export const getCookie = jest.fn().mockReturnValue('dummyCookieValue');
