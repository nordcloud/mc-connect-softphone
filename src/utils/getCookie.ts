import { APIGatewayProxyEventV2 } from 'aws-lambda';
import cookie from 'cookie';

export function getCookie(event: APIGatewayProxyEventV2, cookieName: string) {
  if (!event.cookies || !event.cookies.length) {
    return undefined;
  }

  const cookies = cookie.parse(event.cookies.join(';'));

  return cookies[cookieName] as string | undefined;
}
