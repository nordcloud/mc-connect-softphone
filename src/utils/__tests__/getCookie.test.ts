import cookie from 'cookie';

import { getCookie } from '../getCookie';

jest.mock('cookie', () => ({
  parse: jest.fn().mockReturnValue({
    dummyCookieName: 'dummyCookieValue',
  }),
}));

const args = [{ cookies: ['dummyCookie'] }, 'dummyCookieName'] as Parameters<
  typeof getCookie
>;

describe('getCookie', () => {
  it('calls cookie.parse with correct args', () => {
    getCookie(...args);
    expect(jest.mocked(cookie.parse).mock.calls).toMatchSnapshot();
  });

  it('returns a correct value', () => {
    expect(getCookie(...args)).toMatchSnapshot();
  });

  it('returns a correct value when event.cookies is missing', () => {
    const argsClone = structuredClone(args);
    delete argsClone[0].cookies;
    expect(getCookie(...argsClone)).toMatchSnapshot();
  });
});
