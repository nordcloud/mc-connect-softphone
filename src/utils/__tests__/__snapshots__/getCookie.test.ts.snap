// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCookie calls cookie.parse with correct args 1`] = `
[
  [
    "dummyCookie",
  ],
]
`;

exports[`getCookie returns a correct value 1`] = `"dummyCookieValue"`;

exports[`getCookie returns a correct value when event.cookies is missing 1`] = `undefined`;
